<launch>

    <arg name='goal_image_dir' default='$(find robot_orienteering)/data/delta_demo_images' />

    <arg name='config_dir_path' default='$(find robot_orienteering)/config' />

    <arg name='local_model_type' default='nomad' doc='gnm, vint, nomad' />
    <arg name='local_model_path' default='$(find robot_orienteering)/data/onnx_models' />

    <arg name='global_model_path' default='$(find robot_orienteering)/data/onnx_models/distance_segment.onnx' />
    <arg name='map_path' default='$(find robot_orienteering)/data/maps' />
    <arg name='map_name' default='Delta' />

    <arg name='fps' default='15' />
    <arg name='timer_frequency' default='10' />
    <arg name='record_video' default='' />

    <arg name='nomad_conditioning_threshold' default='3' />
    <arg name='gps_conditioning_threshold' default='10' />  
    <arg name='goal_heuristic' default='euclidean' doc='euclidean, global_planner, hybrid' />     

    <arg name='use_gps' default='True' />
    <arg name='use_global_planner' default='False' />
    <arg name='use_laser' default='True' />
    <arg name='use_nomad' default='False' />

    <arg name='node_name' default='jackal_drive.py' />     

    <node pkg="robot_orienteering" type="$(arg node_name)" name="inference" required="true" output="screen">
        
        <param name="goal_image_dir" value="$(arg goal_image_dir)" />

        <param name="config_dir_path" value="$(arg config_dir_path)" />

        <param name="local_model_type" value="$(arg local_model_type)" />
        <param name="local_model_path" value="$(arg local_model_path)" />

        <param name="global_model_path" value="$(arg global_model_path)" />
        <param name="map_path" value="$(arg map_path)" />
        <param name="map_name" value="$(arg map_name)" />
        
        <param name="fps" value="$(arg fps)" />
        <param name="record_video" value="$(arg record_video)" />
        <param name="timer_frequency" value="$(arg timer_frequency)" />
        
        <param name="base_link_frame" value="base_link" />
        <param name="gps_frame" value="gps_frame" />
        <param name="left_camera_frame" value="zed_left_camera_frame" />
        <param name="left_camera_optical_frame" value="zed_left_camera_optical_frame" />

        <param name="nomad_conditioning_threshold" value="$(arg nomad_conditioning_threshold)" />
        <param name="gps_conditioning_threshold" value="$(arg gps_conditioning_threshold)" />
        <param name="goal_heuristic" value="$(arg goal_heuristic)" />
        
        <param name="use_gps" value="$(arg use_gps)" />
        <param name="use_global_planner" value="$(arg use_global_planner)" />
        <param name="use_laser" value="$(arg use_laser)" />
        <param name="use_nomad" value="$(arg use_nomad)" />

    </node>
    
    <include file="$(find robot_orienteering)/launch/depth_to_laser_scan.launch" />
    
</launch>